src/main/java/com/ericsson/commonlibrary/proxy/InterceptableProxyFactory.java

```diff
@@ -122,7 +122,12 @@ void setSuperclass(Class<?> classToIntercept) {
             }
         }
         try {
-            return cc.toClass();
+        	double javaSpecVersion = Double.parseDouble(System.getProperty("java.specification.version"));
+        	if (javaSpecVersion > 1.8) {
+        		return cc.toClass(javaBean);
+        	} else {        		
+        		return cc.toClass(Thread.currentThread().getContextClassLoader(), javaBean.getProtectionDomain());
```

Comment:
wouldn't cc.toClass(Thread.currentThread().getContextClassLoader(), javaBean.getProtectionDomain()); work with any java version, meaning that the if statement could be skipped.

Reply:
> 
> 
> wouldn't cc.toClass(Thread.currentThread().getContextClassLoader(), javaBean.getProtectionDomain()); work with any java version, meaning that the if statement could be skipped.

This API can not give neighbor info, thus lead to on Java11 it still invoke the old APIs then give warnings as mentioned.
https://github.com/jboss-javassist/javassist/blob/master/src/main/javassist/util/proxy/DefineClassHelper.java#L52